apps/browser/lib/hooks/useEvoService.ts

```diff
@@ -186,6 +188,8 @@ export const useEvoService = (
         await createChat(chatId);
         await handleChatIdChange(chatId);
         onCreateChat(chatId);
+      } else {
+        await updateChatTitle({ chatId, title: goal })
```

Comment:
This will replace any name that was given prior, either from the user's first goal or from a rename. We should ideally only update name on first user goal, and then respect anything a user renames to in the future.